### ISSUE TITLE
quarkus-logging-json not working 1.3.0

### ISSUE THREAD

**Issue by tobiasstadler** [quarkusio/quarkus#7469]
**Describe the bug**
After Upgrading quarkus to 1.3.0.Alpha2 quarkus-logging-json fails in dev mode:
```
2020-02-27 22:21:00,966 ERROR [io.qua.dev.DevModeMain] (main) Failed to start Quarkus: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:232)
	at io.quarkus.runtime.Application.start(Application.java:89)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at io.quarkus.runner.bootstrap.StartupActionImpl.run(StartupActionImpl.java:95)
	at io.quarkus.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:60)
	at io.quarkus.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:236)
	at io.quarkus.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:39)
	at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:130)
	at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:83)
	at io.quarkus.dev.DevModeMain.start(DevModeMain.java:104)
	at io.quarkus.dev.DevModeMain.main(DevModeMain.java:46)
Caused by: java.lang.NoClassDefFoundError: javax/json/Json
	at org.jboss.logmanager.formatters.JsonFormatter.<init>(JsonFormatter.java:60)
	at io.quarkus.logging.json.runtime.LoggingJsonRecorder.initializeJsonLogging(LoggingJsonRecorder.java:17)
	at io.quarkus.deployment.steps.LoggingJsonSteps$setUpFormatter13.deploy_0(LoggingJsonSteps$setUpFormatter13.zig:76)
	at io.quarkus.deployment.steps.LoggingJsonSteps$setUpFormatter13.deploy(LoggingJsonSteps$setUpFormatter13.zig:36)
	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:90)
	... 13 more
Caused by: java.lang.ClassNotFoundException: javax.json.Json
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:602)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	... 18 more
```

**Expected behavior**
quarkus starts and logs in json format

**Actual behavior**
quarkus does not start

**To Reproduce**
1. Upgrade quarkus to 1.3.0.Alpha2
2. Add the quarkus-logging-json extension
3. run quarkus in dev mode (mvn quarkus:dev)

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: Darwin Tobiass-MacBook-Pro.local 19.3.0 Darwin Kernel Version 19.3.0: Thu Jan  9 20:58:23 PST 2020; root:xnu-6153.81.5~1/RELEASE_X86_64 x86_64
 - Output of `java -version`: OpenJDK 64-Bit Server VM (build 13.0.2+8, mixed mode, sharing)
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: 1.3.0.Alpha2

**Comment by gsmet** [comment]
@stuartwdouglas looks like a new CL issue.

**Comment by jaikiran** [comment]
The `org.jboss.logmanager.formatters.JsonFormatter` which is used in the recorder of `quarkus-logging-json` extension, is a class that's part of the `org.jboss.logmanager:jboss-logmanager-embedded` artifact/jar. It so happens that this specific artifact is marked as "parentFirst" here[1]. So when Quarkus runtime classloader tries to load the `JsonFormatter` it gives preference to the parent classloader (to see if it can load that class). It also happens to be the case that this `org.jboss.logmanager:jboss-logmanager-embedded` is an entry in the `META-INF/MANIFEST.MF` of the application's dev mode jar (which is used to launch dev mode) and hence the Java application launch classloader has this available in its classpath and when Quarkus asks this classloader to load `JsonFormatter`, it readily does that. However, this `JsonFormatter` class has a dependency on classes that belong to `javax.json` package. This package is provided by the `org.glassfish:jakarta.json:jar:1.1.6` artifact. This artifact however isn't part of the Java applicaiton launch classloader (it however is, rightly, part of the Quarkus runtime classloader) and hence when it tries to load `JsonFormatter` it doesn't find these `javax.json.*` classes.

Long story short - the jboss-logmanager-embedded which is visible to the Java launch classloader doesn't have some of its dependencies available in that classloader. I can't think of a clean way to get past this, other than maybe adding this artifact to the "parentFirst" list too (however, that won't solve this cleanly - for example what happens if jboss-logmanager-embedded depends on some other similar dependencies, do we keep adding them statically?).

[1] https://github.com/quarkusio/quarkus/blob/master/core/runtime/pom.xml#L109
